elide-core/src/test/java/com/yahoo/elide/core/security/AggregationStorePermissionExecutorTest.java

```diff
@@ -0,0 +1,191 @@
+        elideSettings = new ElideSettingsBuilder(null)
+                .withEntityDictionary(dictionary)
+                .build();
+    }
+
+    @Test
+    public void testUserPermissions() {
+        @Entity
+        @Include
+        @Value
+        @ReadPermission(expression = "user none or filter check")
+        class Model {
+            String filterDim;
+            long metric;
+            long metric2;
+
+            @ReadPermission(expression = "user all")
+            public String getFilterDim() {
+                return filterDim;
+            }
+
+            @ReadPermission(expression = "user none")
+            public long getMetric() {
+                return metric;
+            }
+        }
+
+        com.yahoo.elide.core.RequestScope scope = bindAndgetRequestScope(Model.class);
+        PermissionExecutor executor = scope.getPermissionExecutor();
+
+        // evaluated expression = user all
+        Assertions.assertEquals(
+                ExpressionResult.PASS,
+                executor.checkUserPermissions(ClassType.of(Model.class), ReadPermission.class, "filterDim"));
+
+
+        // evaluated expression = user none -> ForbiddenAccess
+        Assertions.assertThrows(
+                ForbiddenAccessException.class,
+                () -> executor.checkUserPermissions(ClassType.of(Model.class), ReadPermission.class, "metric"));
+
+
+        // evaluated expression = (user none or filter check)
+        Assertions.assertEquals(
+                ExpressionResult.DEFERRED,
+                executor.checkSpecificFieldPermissions(
+                        new PersistentResource(new Model("dim1", 0, 1), "1", scope),
+                        null, ReadPermission.class, "metric2"));
+
+
+        // evaluated expression = (user none or filter check) AND (user all OR user none)
+        Assertions.assertEquals(
+                ExpressionResult.DEFERRED,
+                executor.checkUserPermissions(ClassType.of(Model.class), ReadPermission.class, new HashSet<>(Arrays.asList("filterDim", "metric"))));
+
+        // evaluated expression = (user all OR user none)
+        Assertions.assertEquals(
+                ExpressionResult.PASS,
+                executor.checkPermission(ReadPermission.class,
+                        new PersistentResource(new Model("dim1", 0, 1), "1", scope),
+                        new HashSet<>(Arrays.asList("filterDim", "metric"))));
+
+
+    }
+
+    @Test
+    public void filterTest() {
+        @Entity
+        @Include
+        @Value
+        @ReadPermission(expression = "user none or filter check")
+        class Model1 {
+            String filterDim;
+            long metric;
+        }
+
+        com.yahoo.elide.core.RequestScope scope = bindAndgetRequestScope(Model1.class);
+        PermissionExecutor executor = scope.getPermissionExecutor();
+
+        FilterExpression expression = executor.getReadPermissionFilter(ClassType.of(Model1.class), Collections.emptySet()).orElse(null);
```

Comment:
To make these tests realistic, the set of requested fields should not be empty - but rather set to the all the fields.